src/Cli/dotnet/Commands/Tool/ToolCommandSpecCreator.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.CommandLine;
+using System.Text;
+using Microsoft.DotNet.Cli.CommandFactory;
+using Microsoft.DotNet.Cli.CommandFactory.CommandResolution;
+using Microsoft.DotNet.Cli.ToolManifest;
+using Microsoft.DotNet.Cli.ToolPackage;
+using Microsoft.DotNet.Cli.Utils;
+
+namespace Microsoft.DotNet.Cli.Commands.Tool
+{
+    internal class ToolCommandSpecCreator
+    {
+        public static CommandSpec CreateToolCommandSpec(string toolName, string toolExecutable, string toolRunner, bool allowRollForward, IEnumerable<string> commandArguments)
+        {
+            if (toolRunner == "dotnet")
+            {
+                if (allowRollForward)
+                {
+                    commandArguments = ["--allow-roll-forward", .. commandArguments];
+                }
+
+                return MuxerCommandSpecMaker.CreatePackageCommandSpecUsingMuxer(
+                    toolExecutable,
+                    commandArguments);
+            }
+            else if (toolRunner == "executable")
```

Comment:
nit: Redundant `else` usage (both this else-if and the plain else).

Reply:
It's redundant but seems a bit clearer to me like this.  If you get rid of the `else`s you have to pay attention to the return statements inside the blocks to understand the overall flow.